main/src/library/Boxable.flix

```diff
@@ -0,0 +1,247 @@
+        case _             => ?bug
+    }
+}
+
+instance Boxable[Int32] {
+    pub def box(x: Int32): Boxed = BoxedInt32(x)
+    pub def unbox(x: Boxed): Int32 = match x {
+        case BoxedInt32(v) => v
+        case _             => ?bug
+    }
+}
+
+instance Boxable[Int64] {
+    pub def box(x: Int64): Boxed = BoxedInt64(x)
+    pub def unbox(x: Boxed): Int64 = match x {
+        case BoxedInt64(v) => v
+        case _             => ?bug
+    }
+}
+
+instance Boxable[Float32] {
+    pub def box(x: Float32): Boxed = BoxedFloat32(x)
+    pub def unbox(x: Boxed): Float32 = match x {
+        case BoxedFloat32(v) => v
+        case _               => ?bug
+    }
+}
+
+instance Boxable[Float64] {
+    pub def box(x: Float64): Boxed = BoxedFloat64(x)
+    pub def unbox(x: Boxed): Float64 = match x {
+        case BoxedFloat64(v) => v
+        case _               => ?bug
+    }
+}
+
+instance Boxable[BigInt] {
+    pub def box(x: BigInt): Boxed = {
+        let value = x as ##java.lang.Object;
+        let compare = (o1, o2) -> Order.compare(o1 as BigInt, o2 as BigInt);
+        let toString = o -> ToString.toString(o as BigInt);
+        BoxedObject(value, compare, toString)
+    }
+    pub def unbox(x: Boxed): BigInt = match x {
+        case BoxedObject(v, _, _) => v as BigInt
+        case _                    => ?bug
+    }
+}
+
+instance Boxable[String] {
+    pub def box(x: String): Boxed = {
+        let value = x as ##java.lang.Object;
+        let compare = (o1, o2) -> Order.compare(o1 as String, o2 as String);
+        let toString = o -> ToString.toString(o as String);
+        BoxedObject(value, compare, toString)
+    }
+    pub def unbox(x: Boxed): String = match x {
+        case BoxedObject(v, _, _) => v as String
+        case _                    => ?bug
+    }
+}
+
+instance Boxable[(a, b)] with Boxable[a], Boxable[b] {
+    pub def box(x: (a, b)): Boxed = {
+        let value = x as ##java.lang.Object;
+        let compare = (o1, o2) -> Order.compare(o1 as (a, b), o2 as (a, b));
+        let toString = o -> ToString.toString(o as (a, b));
+        BoxedObject(value, compare, toString)
+    }
+    pub def unbox(x: Boxed): (a, b) = match x {
+        case BoxedObject(v, _, _) => v as (a, b)
+        case _                    => ?bug
+    }
+}
+
+instance Boxable[(a, b, c)] with Boxable[a], Boxable[b], Boxable[c] {
+    pub def box(x: (a, b, c)): Boxed = {
+        let value = x as ##java.lang.Object;
+        let compare = (o1, o2) -> Order.compare(o1 as (a, b, c), o2 as (a, b, c));
+        let toString = o -> ToString.toString(o as (a, b, c));
+        BoxedObject(value, compare, toString)
+    }
+    pub def unbox(x: Boxed): (a, b, c) = match x {
+        case BoxedObject(v, _, _) => v as (a, b, c)
+        case _                    => ?bug
+    }
+}
+
+// TODO: Benj: Add for Tuples 3, 4, 5.
+
+// TODO: Benj: Add Option
+// TODO: Benj: Add List
+// TODO: Benj: Add Nel
+// TODO: Benj: Add Set/Map
```

Comment:
It would be nice if you could help fix these TODOs.

Reply:
Most of these have already been implemented.

Reply:
Fixed.